syntax = "proto3";

package book;

import "google/protobuf/empty.proto";

service BookController {
    rpc List(BookListRequest) returns (stream Book) {}
    rpc Create(Book) returns (Book) {}
    rpc Retrieve(BookRetrieveRequest) returns (Book) {}
    rpc Update(Book) returns (Book) {}
    rpc Destroy(Book) returns (google.protobuf.Empty) {}
}

message Book {
    int32 id = 1;
    string title = 2;
    int32 author = 3;
    string summary = 4;
    string isbn = 5;
    repeated int32 genre = 6;
    int32 language = 7;
    string image = 8;
}

message BookListRequest {
}

message BookRetrieveRequest {
    int32 id = 1;
}
